eccairs-to-rdf/src/main/java/com/github/psiotwo/eccairs/rdf/TaxonomyService.java

```diff
@@ -0,0 +1,27 @@
+package com.github.psiotwo.eccairs.rdf;
+
+import com.github.psiotwo.eccairs.core.model.EccairsValue;
+
+import java.util.List;
+
+/**
+ * Provides access to the ECCAIRS taxonomy service.
+ */
+public interface TaxonomyService {
+
+    /**
+     * Checks whether an attribute with the specified ECCAIRS ID has a hierarchical value list.
+     *
+     * @param attributeId Attribute ECCAIRS ID
+     * @return {@code} true when attribute value list has more than one level, {@code false} otherwise
+     */
+    boolean hasHierarchicalValueList(int attributeId);
+
+    /**
+     * Gets a presumably hierarchical value list of an attribute with the specified ECCAIRS ID.
+     *
+     * @param attributeId Attribute ECCAIRS ID
+     * @return List of top level attribute values
+     */
+    List<EccairsValue> getValueList(int attributeId);
```

Comment:
How long the execution actually takes?

Some value lists might be shared across attributes, so the execution time can be shortened by providing attribute->valuelist mapping and then valuelist->List<EccairsValue> (but maybe you have it directly in the impl., hidden from the API?)

Reply:
The problem is that the ECCAIRS taxonomy service uses internal identifiers that have nothing in common with the taxonomy identifiers. Internally, the implementation remembers this mapping for already visited attributes.
Execution takes up to tens of seconds for large/deeply nested value lists (like Aircraft manufacturer/model).

Reply:
OK, probably we do not have enough info in data for this optimization.